theano/gof/vm.py

```diff
@@ -717,7 +717,13 @@ def make_vm(self, nodes, thunks,
                 updated_vars
                 ):
 
-        pre_call_clear = [storage_map[v] for v in self.no_recycling]
+        # Only put in pre_call clear variables that have not been computed
+        # inplace
+        pre_call_clear = [storage_map[v] for v in self.no_recycling
+                          if v.owner is None or
+                          not hasattr(v.owner.op, 'destroy_map') or
+                          v.owner.outputs.index(v) not in
+                          v.owner.op.destroy_map.keys()]
```

Comment:
I'm not 100% sure to be honest.  But I would guess that the split is because it's the linker's job to create it and both of those files have a linker.  Maybe you could move that to a shared function?